src/java/org/apache/cassandra/config/Converter.java

```diff
@@ -0,0 +1,204 @@
+/*
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+package org.apache.cassandra.config;
+
+/**
+ * An interface to support conversions for backward compatibility with the old cassandra.yaml where duration, bit rate and
+ * data storage configuration parameters were provided only by value and the expected unit was part of the configuration
+ * parameter name. (CASSANDRA-15234)
+ * It is important to be noted that this converter is not intended to be used when we don't change name of a configuration
+ * parameter but we want to add unit. This would always default to the old value provided without a unit at the moment.
+ * in case this functionality is needed at some point, please, raise a ticket.
+ */
+public interface Converter<Original, Current>
```

Comment:
I know there was some commentary in the ticket to the effect that we couldn't use an `enum` for the converters because of some concern around generics. I'm just not immediately able to figure out what that is. There are a number of compiler warnings in `YamlConfigurationLoader` at the moment around `Converter`, but that might be another issue...